### DISCUSSION TITLE
Can Snap actively trigger Metamask?

### DISCUSSION THREAD

**Discussion by LaplaceMan** [MetaMask/snaps#1765] [Q&A]
Can Snap initiate a transaction and then have Metamsk sign it?

It seems not in the current stage, is this feature in the future plans, I'm sure it could make Snap even more powerful. 
There really is a lot of things in my design that need to be actively interacting with Metamask, rather than having Snap always be passive.

**Comment by Montoya** [comment]
It's not possible in the current API, and I can't say by when it will be possible. We have various API features that we are currently working on, you can see what's in the pipeline in our Snaps Improvement Proposals repo: https://metamask.github.io/SIPs/
